Lib/datetime.py

```diff
@@ -454,20 +454,24 @@ def __new__(cls, days=0, seconds=0, microseconds=0,
         return self
 
     def __repr__(self):
+        arg_str = ""
+
+        if self._days:
+            arg_str += "days=%d" % self._days
+
+        if self._seconds or (not self._days and not self._microseconds):
+            if arg_str != "":
+                arg_str += ", "
+            arg_str += "seconds=%d" % self._seconds
+
         if self._microseconds:
-            return "%s.%s(%d, %d, %d)" % (self.__class__.__module__,
-                                          self.__class__.__qualname__,
-                                          self._days,
-                                          self._seconds,
-                                          self._microseconds)
-        if self._seconds:
-            return "%s.%s(%d, %d)" % (self.__class__.__module__,
-                                      self.__class__.__qualname__,
-                                      self._days,
-                                      self._seconds)
-        return "%s.%s(%d)" % (self.__class__.__module__,
+            if arg_str != "":
+                arg_str += ", "
+            arg_str += "microseconds=%d" % self._microseconds
```

Comment:
Concatenate strings using += is inefficient. Moreover, you have to handle comma manually. Please rewrite this code using a list (args = [] ... args.append(...)), and then join it using: ', '.join(args).